elogia_sale/models/elogia_sale.py

```diff
@@ -226,12 +231,86 @@ def calc_amount_total(self):
                     val_tmp = record.invoice_provision / record.currency_id.rate
                 else:
                     val_tmp = (record.invoice_provision / record.currency_id.rate) * record.control_id.currency_id.rate
+                    if record.currency_id == record.control_id.company_id.currency_id:
+                        val_tmp = record.invoice_provision * record.control_id.currency_id.rate
             record.invoice_provision_ml = val_tmp
 
-    @api.model
     def create_purchase_order(self):
-        for record in self:
-            pass
+        list_supplier = [line for line in self if line.state == 'no_process' and line.type_payment != 'client'
+                         and line.control_id.state in ['pending', 'sale']]
+        if list_supplier:
+            line_by_supplier = self._group_line_by_supplier(list_supplier)
+            if line_by_supplier:
+                for supplier_key, supplier_value in line_by_supplier.items():
+                    line_by_currency = self._group_line_by_currency(supplier_value)
+                    if line_by_currency:
+                        for currency_key, currency_value in line_by_currency.items():
+                            self.create_purchase(currency_value, supplier_key)
+                            for item in currency_value:
+                                item.state = 'process'
+        for control in set([item.control_id for item in list_supplier]):
```

Comment:
este for lo puedes cambiar por:
for control in list_supplier.mapped('control_id'):

Reply:
Listo